src/android/com/fivestars/mtab/plugin/RestartAppPlugin.java

```diff
@@ -31,7 +31,13 @@
     public boolean execute(String action, JSONArray args, final CallbackContext callbackContext) throws JSONException {
 
         if (EXECUTE_RESTART.equals(action)) {
-            return restartApp();
+            if (args.length() == 2) {
+                long delay = args.getLong(0);
+                boolean exit = args.getBoolean(1);
+                return restartApp(delay, exit);
+            } else {

```

Comment:
You could do `final` in front of the locals. Not important.
Also you don't need the `else` since the `if` exits with a return.